main.tf

```diff
@@ -0,0 +1,138 @@
+/**
+ * Creates a Lambda function with associated role and policies that
+ * will run the packer-janitor tool to get rid of orphaned Packer
+ * instances and their associated resources.
+ *
+ * Creates the following resources:
+ *
+ * * Lambda function
+ * * IAM role and policies to describe and delete instances, keypairs,
+ *   and security groups, as well as write messages to Cloudwatch Logs
+ * * Cloudwatch Logs group
+ * * Cloudwatch Event to regularly run cleanup job
+ *
+ * ## Usage
+ *
+ * ```hcl
+ * module "packer-janitor-lambda" {
+ *   source                 = "trussworks/lambda-packerjanitor/aws"
+ *   packer_resource_delete = "true"
```

Comment:
Should this be a boolean `true` instead of a string?

Reply:
I have a vague recollection that when I tried this before it barfed on that -- that's why the variable is set as a string instead of a boolean in variables.tf. This might be something fixed in tf 0.12, but this is still using 0.11.

Reply:
I bet it was barfing on passing it in as an env var below. At that point you'd need a conditional to pass in a proper string instead:

```hcl
env_var = "${packer_resource_delete ? "true" : "false"}"
```

For me, this would be more "correct". That flag is a boolean and should take a boolean, not a string. It also would protect against someone accidentally mis-typing the string and getting unexpected behavior. Like, does the env var check for the string "true" or the string "false"? What happens if the user accidentally typed "fasle" or "treu"?

Reply:
This is what I get if I try to change that to "bool" type:
```
Error: Error loading /Users/cblkwell/git-repos/terraform-aws-lambda-packerjanitor/variables.tf: Variable 'packer_resource_delete' type must be one of [list, string, map] - 'bool' is not a valid type
```

https://www.terraform.io/docs/configuration-0-11/variables.html specifies list, string, and map as the only valid variable types in 0.11, unfortunately. In 0.12 the number of variable types is greatly expanded, but we have not made that switch yet. :/

Reply:
I should note that packer-janitor's default is to *not* delete and only run in dry run mode, so it should be fail-safe. In addition, *it* expects that variable to be a boolean, so it if it something weird the lambda job will simply fail, rather than achieving sentience and going Skynet on your infrastructure.

Reply:
The type is still string, but you can set it via a bool. It's weird:

https://github.com/trussworks/terraform-aws-alb-web-containers/blob/master/variables.tf#L35

Reply:
Okay, I see what you're saying. I've fixed that in the latest commit.